examples/cross_build/toolchain_packages/toolchain_macos_linux_cross/conanfile.py

```diff
@@ -0,0 +1,89 @@
+import os
+from conan import ConanFile
+from conan.tools.files import get, copy, download
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+
+
+class ArmToolchainPackage(ConanFile):
+    name = "arm-toolchain"
+    version = "13.2"
+
+    license = "GPL-3.0-only"
+    homepage = "https://github.com/messense/homebrew-macos-cross-toolchains/"
+    description = "Conan package for test cross-compilation from Macos to Linux."
+    settings = "os", "arch"
+    package_type = "application"
+
+    def _get_arch(self, arch):
+        if arch in self._archs_aarch64():
+            return "aarch64"
+        elif arch in self._archs_x86_64():
+            return "x86_64"
+
+    def _archs_x86_64(self):
+        return ["x86_64"]
+    
+    def _archs_aarch64(self):
+        return ["armv8", "armv8_32", "armv8.3", "arm64ec"]
+
+    def _get_toolchain(self, target_arch):
+        return f"{self._get_arch(target_arch)}-unknown-linux-gnu"
+    
+    def _get_toolchain_name(self, arch_macos, arch_linux):
+        return f"{arch_linux}-unknown-linux-gnu-{arch_macos}-darwin.tar.gz"
+    
+    def _hash_validator(self, toolchain):
+        d = {
+            "aarch64-unknown-linux-gnu-aarch64-darwin.tar.gz": "a87669a9df908d8d8859849a0f9fc0fb287561a4e449c21dade10663d42d2ccb"
+            "aarch64-unknown-linux-gnu-x86_64-darwin.tar.gz": "6979291e34064583ac8b12a8b6b99ec6829caf22f47bcb68b646365ec9e24690"
+            "x86_64-unknown-linux-gnu-aarch64-darwin.tar.gz": "bb59598afd84b4d850c32031a4fa64c928fb41f8ece4401553b6c23714efbc47"
+            "x86_64-unknown-linux-gnu-x86_64-darwin.tar.gz": "86e28c979e5ca6d0d1019c9b991283f2ab430f65cee4dc1e4bdf85170ff7c4f2"
+        }
+        return d[toolchain]
+
+    def _get_toolchain_url(self):
+        arch_macos = self._get_arch(self.settings.arch)
+        arch_linux = self._get_arch(self.settings_target.arch)
+        toolchain = self._get_toolchain_name(arch_macos, arch_linux)
+        hash_sha256 = self._hash_validator(toolchain)
+        base_url = f"https://github.com/messense/homebrew-macos-cross-toolchains/releases/download/v13.2.0/{toolchain}"
+        return base_url, hash_sha256
+
+    def package_id(self):
+        self.info.settings_target = self.settings_target
+        # We only want the ``arch`` setting
+        self.info.settings_target.rm_safe("os")
```

Comment:
How does the package info end up being? I would have thought that the target OS needs to be represented? I know this was in the other recipe as well, so perhaps this is a question for @czoido !

Reply:
I think that could be because it the `settings_target` for the other toolchain can only be Linux, so it's no relevant.